### PR TITLE
All projects compile, all but WindowStore run without errors

### PR DESCRIPTION
**Pull request by PeterBurke**
INTERIM PULL REQUEST
=====================

This is an interim PR to enable reviewers to see progress and preview.

There will be more PR's to follow, and these will incorporate this PR. I suggest waiting for the last PR before we update the Master. But with Git you can clone this Repo with the PR to your desktop and review.

The current PR addresses an issue where WindowsPhone would not run due to a missing reference, and compile errors in the WindowStore project. Before addressing these compile errors I needed to satisfy myself that the WindowStore projects were linked via NuGet to the correct MvvMCross packages. I was puzzled by the fact that MvvMCross had a WindowStore NuSpec but NuGet decided to link to the WindowsCommon NuSpec. NuGet makes its choices based on the ProjectTypeGuid in the CSPROJ file. I verified that this was correct as per Xamarin documentation. NuGet decided to load MvvmCrossLibraries via the **MvvmCross.HotTuna.MvvmCrossLibraries.3.0.1.nuspec** a fragment of which follows
```
<!-- store -->
<file src="..\bin\Release\Mvx\WindowsStore\Cirrious.MvvmCross.WindowsStore.dll" target="lib\netcore45\Cirrious.MvvmCross.WindowsStore.dll" />
<file src="..\bin\Release\Mvx\WindowsStore\Cirrious.MvvmCross.WindowsStore.pdb" target="lib\netcore45\Cirrious.MvvmCross.WindowsStore.pdb" />

<!-- windows common -->
<file src="..\bin\Release\Mvx\WindowsCommon\Cirrious.MvvmCross.WindowsCommon.dll" target="lib\win81\Cirrious.MvvmCross.WindowsCommon.dll" />
<file src="..\bin\Release\Mvx\WindowsCommon\Cirrious.MvvmCross.WindowsCommon.pdb" target="lib\win81\Cirrious.MvvmCross.WindowsCommon.pdb" />
<file src="..\bin\Release\Mvx\WindowsCommon\Cirrious.MvvmCross.WindowsCommon.dll" target="lib\wpa81\Cirrious.MvvmCross.WindowsCommon.dll" />
<file src="..\bin\Release\Mvx\WindowsCommon\Cirrious.MvvmCross.WindowsCommon.pdb" target="lib\wpa81\Cirrious.MvvmCross.WindowsCommon.pdb" />
```

So NuGet given the ProjectTypeGuid decides to load from:**win81** and not from **netcore45**
This might still be an issue because right now the Samples have no View to load.

In the next PR I will to add a C# view to each sample, plus test projects and prove that they run.

This should be followed by a PR that restructures and renames things. I would like to see the samples using NuGet referencing to the FormsPresenters rather than project referencing.

### CODE REVIEW

File: MvvmCross_Forms_Examples.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.30723.0
+# Visual Studio 14
```
Comment (martijn00) [inline MvvmCross_Forms_Examples.sln]: I am not sure if it is a good idea to use 14 yet. Please use 2013.

File: Samples/Example.WindowStore/Setup.cs [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 using Xamarin.Forms;
+using W = Windows.UI.Xaml.Controls;
+using Cirrious.MvvmCross.WindowsCommon.Views;
 
 namespace Example.WindowStore
 {
     public class Setup : MvxWindowsSetup
     {
-        public Setup(Frame rootFrame) : base((IMvxWindowsFrame)rootFrame)
+        public Setup(W.Frame rootFrame) : base(rootFrame)
```
Comment (martijn00) [inline Samples/Example.WindowStore/Setup.cs]: "W" is a bit of a short name. Could you use something more understandable?

**Comment by martijn00** [comment]
Thanks for this PR. Looks good mostly!

File: MvvmCross_Forms_Examples.sln (hunk 1 above)
Comment (PeterBurke) [inline MvvmCross_Forms_Examples.sln]: Thanks Martijn
My current laptop running VS2013 is broken. Will download VS 2013 from subscriber downloads and set up on a VM. SLN file is very simple so if we are lucky simply changing it to 2013 should work. The real danger would be in VS2013 running older versions of NuSpec etc., but what we have here should be OK.

VS2015 RTM will be available from MSDN subscriber downloads July / 20 and presumably to the public then or shortly after. Presumably you get a free trial period typically 180 days if you do not have MSDN.

**Comment by PeterBurke** [comment]
WindowStore sample throws an exception very early during testing.

As per Xamarin developer notes, you put the following line of code into the App.xaml.cs:

Xamarin.Forms.Forms.Init(e). 

I wrote an example 6 weeks ago and it works fine. But for some reason in this project it throws an E_FAIL COM Exception while attempting to update the resource dictionary. Others have had this error and Xamarin were unable to help. The situation is very opaque i.e. we cannot see past the COM call and E_FAIL is just 1 bit of information.

So what I will do is introduce another Sample called Sample.W81 based on my sample from 6 weeks ago that gets past this point, then gradually add the extras. This may of course ultimately mean removing the Sample.WindowStore project, but we can deal with the naming issues later.

**Comment by PeterBurke** [comment]
Now at the point where Windows.w81 Sample (replacing WindowsStore sample) runs all the way through without exceptions. I created a FirstPage to match the FirstViewModel and it locates it. I am getting a blank screen though. Also need to test the FirstPage in all the other targets.

I might issue a progressive PR later today. However if you can wait a day, I will also attempt to down convert from VS2015 to VS2013. Hopefully there will be no change to c# files or csproj files. Currently I have not had a situation where loading an MvvMCross solution has triggered an upgrade, so the VS version issue might be simply a case of downgrading the version numbers in the SLN file before issuing the PR.

**Comment by martijn00** [comment]
Yes we can wait a bit. I think we just want this to be stable before the release of 4.0.0

**Pull request closed without merging**